src/storage/httpStore.ts

```diff
@@ -20,6 +21,14 @@ export class HTTPStore implements AsyncStore<ArrayBuffer> {
     async getItem(item: string) {
         const url = new URL(item, this.url).href;
         const value = await fetch(url);
+
+        if (value.status === 404) {
+            // Item is not found
+            throw new KeyError(item);
+        } else if (value.status !== 200) {
```

Comment:
We may need to add a few more in the future (e.g. *202 found* may be fine), but I think we'll find out automatically through issues in the future ^^

Reply:
Completely agree here. This PR was to cut down on requests, and allow for better error handling in the future. As it was before, if `this.containsItem(cKey) !== true` fetching the chunk was skipped all together and the code for `HttpStore.containsItem` is only checking for 200 status:

```javascript
async containsItem(item: string): Promise<boolean> {
        const url = new URL(item, this.url).href;
        const value = await fetch(url);
        return value.status === 200;
}
```